### PR TITLE
Advance the address instead of repeating for memory commands

### PR DESCRIPTION
**Pull request by wbowling**
### Description/Motivation/Screenshots ###
Currently when running commands like `telescope` or `hexdump` and you press return a second time the same command is run again. This PR adds a boolean property called `repeat` to `GenericCommand` allowing commands handlers to know whether the user had repeated the command by just pressing return.

This allows for behaviour similar to gdbs `x` where you can keep scanning through memory by just pressing return.

Currently this PR used the new property in `DereferenceCommand` and `HexdumpCommand` to advance the start address on a repeat.

image

### How Has This Been Tested? ###

| Architecture | Yes/No                   | Comments               |
|--------------|:------------------------:|------------------------|
| x86-32       | :heavy_multiplication_x: |                         |
| x86-64       | :heavy_check_mark: |                        |
| ARM          | :heavy_multiplication_x: |                        |
| AARCH64      | :heavy_multiplication_x: |                        |
| MIPS         | :heavy_multiplication_x: |                        |
| POWERPC      | :heavy_multiplication_x: |                        |
| SPARC        | :heavy_multiplication_x: |                        |

### Checklist ###

- [x] My code follows the code style of this project.
- [x] My change includes a change to the documentation, if required.
- [x] I have read and agree to the **CONTRIBUTING** document.

---
This change is [Reviewable](https://reviewable.io/reviews/hugsy/gef/333)

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -7469,6 +7482,7 @@ class HexdumpCommand(GenericCommand):
 
     def __init__(self):
         super(HexdumpCommand, self).__init__(complete=gdb.COMPLETE_LOCATION)
+        self.repeat_count = 0
```
Comment (Grazfather) [inline gef.py]: This is fine for now, but I might consider adding this in the `GenericCommand`. Maybe set the `repeat_count` in `invoke`. That way all commands have `repeat_count` if they want it.

**Comment by Grazfather** [comment]
I wouldn't worry about it.

File: gef.py [hunk 2]
```diff
@@ -7536,8 +7553,10 @@ def _hexdump(self, start_addr, length, arrange_as):
         fmt_pack = endianness + r
         lines = []
 
-        i = 0
-        while i < length:
+        i = 0 + self.repeat_count * length
```
Comment (Grazfather) [inline gef.py]: Why not calculate the start address _outside_ of this function (e.g. near where you do the above `fmt == "byte"`. That way this function doesn't need `self` at all.

Comment (wbowling) [reply gef.py]: I wanted to keep the offsets valid, if we change `start_addr` then they reset to 0 on repeat. Could add a new param to this function instead though?

Comment (Grazfather) [reply gef.py]: Ah! Yes fair point. That might make sensei `offset=None`, and if it's set, use it as the offset, otherwise use `start_address`

Comment (Grazfather) [reply gef.py]: `start_offset` or `base`, your call

**Comment by Grazfather** [comment]
LGTM

**Comment by hugsy** [comment]
I love this ! GG @wbowling :tada:

**Pull request merged**